### ISSUE TITLE
ItemUploadOptions not honored

### ISSUE THREAD

**Issue by leepfrog-ger** [OneDrive/onedrive-explorer-win#17]
When uploading a file and specifing the option to Fail on a name conflict this setting seems to be ignored. The already existing file is silently overwritten.

```
var opt = new ItemUploadOptions() { NameConflict = NameConflictBehavior.Fail };
var result = await Connection.PutNewFileToParentItemAsync(folder.ItemReference(), "upload2", fStr, opt);
```

**Comment by rgregg** [comment]
That's true. I didn't get to implementing that in this sample. It should be pretty easy to add support for though, if you want to submit a pull request.

**Comment by leepfrog-ger** [comment]
Hi gregg,

thanks for the reply. Unfortunatly I am not that experienced with programming in general so this would probably result in a lot of try and error and spaghetti code ;)

Regarding the actual issue: I know that this is just a PoC/demo but as this repository is linked on the official microsoft api homepage I think it would be better to throw a NotImplemented Exception or something like this for an operation that seems to be implemented but is not as this would result in data loss.
